scripts/anonymize.sql

```diff
@@ -41,27 +90,102 @@ TRUNCATE djcelery_periodictask;
+DROP TABLE IF EXISTS teamwork_policy_groups;
+DROP TABLE IF EXISTS teamwork_policy_permissions;
+DROP TABLE IF EXISTS teamwork_policy_users;
+DROP TABLE IF EXISTS teamwork_role;
+DROP TABLE IF EXISTS teamwork_role_permissions;
+DROP TABLE IF EXISTS teamwork_role_users;
+DROP TABLE IF EXISTS teamwork_team;
+DROP TABLE IF EXISTS threadedcomments_freethreadedcomment;
+DROP TABLE IF EXISTS threadedcomments_testmodel;
+DROP TABLE IF EXISTS threadedcomments_threadedcomment;
+
+-- To be dropped in bug 1180208, August 2015
+DROP TABLE IF EXISTS user_profiles;
+
+UPDATE account_emailaddress SET
+    email = CONCAT(MD5(CONCAT(email, @common_hash_secret)), '@example.com');
 
 UPDATE auth_user SET
     -- username left alone, because it's public info
     password = '!',
     email = CONCAT(MD5(CONCAT(email, @common_hash_secret)), '@example.com'),
-    first_name = ROUND(RAND()*1000000),
-    last_name = ROUND(RAND()*1000000);
-    location = ROUND(RAND()*1000000),
-    homepage = CONCAT('http://example.com/', ROUND(RAND()*1000000));
+    first_name = '',
+    last_name = '',
+    homepage = '';
+UPDATE auth_user SET
+    title = CONCAT("Title ", MD5(CONCAT(fullname, @common_hash_secret)))
+    WHERE title != "";
+UPDATE auth_user SET
+    fullname = CONCAT("Fullname ", MD5(CONCAT(fullname, @common_hash_secret)))
+    WHERE fullname != "";
+UPDATE auth_user SET
+    organization = CONCAT("Organization ", MD5(CONCAT(organization, @common_hash_secret)))
+    WHERE organization != "";
+UPDATE auth_user SET
+    irc_nickname = CONCAT("irc_", MD5(CONCAT(irc_nickname, @common_hash_secret)))
+    WHERE irc_nickname != "";
+UPDATE auth_user SET
+    location = CONCAT("Location ", MD5(CONCAT(location, @common_hash_secret)))
+    WHERE location != "";
+UPDATE auth_user SET
+    bio = CONCAT("Bio ", MD5(CONCAT(bio, @common_hash_secret)))
+    WHERE bio != "";
+UPDATE auth_user SET
+    facebook_url = CONCAT("https://facebook.com/", MD5(CONCAT(facebook_url, @common_hash_secret)))
+    WHERE facebook_url != "";
+UPDATE auth_user SET
+    github_url = CONCAT("https://github.com/", MD5(CONCAT(github_url, @common_hash_secret)))
+    WHERE github_url != "";
+UPDATE auth_user SET
+    linkedin_url = CONCAT("https://www.linkedin.com/in/", MD5(CONCAT(linkedin_url, @common_hash_secret)))
+    WHERE linkedin_url != "";
+UPDATE auth_user SET
+    mozillians_url = CONCAT("https://mozillians.org/u/", MD5(CONCAT(mozillians_url, @common_hash_secret)))
+    WHERE mozillians_url != "";
+UPDATE auth_user SET
+    stackoverflow_url = CONCAT("https://stackoverflow.com/users/1/", MD5(CONCAT(stackoverflow_url, @common_hash_secret)))
+    WHERE stackoverflow_url != "";
+UPDATE auth_user SET
+    twitter_url = CONCAT("https://twitter.com/", MD5(CONCAT(twitter_url, @common_hash_secret)))
+    WHERE twitter_url != "";
+UPDATE auth_user SET
+    website_url = CONCAT("https://example.com/", MD5(CONCAT(website_url, @common_hash_secret)))
+    WHERE website_url != "";
```

Comment:
Ah, I see. This preserves blank fields as blank - i.e., it doesn't put an md5 hash into the field if it was blank.